plugins/aea-cli-benchmark/aea_cli_benchmark/case_acn_communication/case.py

```diff
@@ -0,0 +1,208 @@
+            mailbox=True,
+            delegate=True,
+            mailbox_port=DEFAULT_MAILBOX_PORT + 2,
+            delegate_port=DEFAULT_DELEGATE_PORT + 2,
+        )
+        await relay_node2.connect()
+        connections.append(relay_node2)
+
+        relay_node1_multiaddr = relay_node1.node.multiaddrs[0]
+        relay_node2_multiaddr = relay_node2.node.multiaddrs[0]
+        await asyncio.sleep(1)
+        con1 = con_maker(
+            port=DEFAULT_NODE_PORT + 10,
+            entry_peer=relay_node1_multiaddr,
+            mailbox_port=DEFAULT_MAILBOX_PORT + 1,
+            delegate_port=DEFAULT_DELEGATE_PORT + 1,
+            pub_key=relay_node1.node.pub,
+        )
+        await con1.connect()
+        connections.append(con1)
+
+        con2 = con_maker(
+            port=DEFAULT_NODE_PORT + 20,
+            entry_peer=relay_node2_multiaddr,
+            mailbox_port=DEFAULT_MAILBOX_PORT + 2,
+            delegate_port=DEFAULT_DELEGATE_PORT + 2,
+            pub_key=relay_node2.node.pub,
+        )
+        await con2.connect()
+        connections.append(con2)
+
+        envelope = make_envelope(con1.address, con2.address)
+
+        with time_measure() as tm:
+            await con1.send(envelope)
+            envelope = await con2.receive()
+        first_time = tm.time
+
+        with time_measure() as tm:
+            await con1.send(envelope)
+            envelope = await con2.receive()
+
+        second_time = tm.time
+
+        return first_time, second_time
+
+    finally:
+        for con in reversed(connections):
+            await con.disconnect()
+
+
+def run(connection: str, run_times: int = 10) -> List[Tuple[str, Union[int, float]]]:
+    """Check construction time and memory usage."""
+    logging.basicConfig(level=logging.CRITICAL)
+    cwd = os.getcwd()
+    try:
+        if connection == "p2pnode":
+
+            def con_maker(
+                port: int,
+                entry_peer: str,
+                mailbox_port: int,
+                delegate_port: int,
+                pub_key: str,
+            ):
+                return _make_libp2p_connection(".", port=port, entry_peers=[entry_peer])
+
+        elif connection == "client":
+
+            def con_maker(
+                port: int,
+                entry_peer: str,
+                mailbox_port: int,
+                delegate_port: int,
+                pub_key: str,
+            ):
+                return _make_libp2p_client_connection(
+                    peer_public_key=pub_key, data_dir=".", node_port=delegate_port
+                )
+
+        elif connection == "mailbox":
+
+            def con_maker(
+                port: int,
+                entry_peer: str,
+                mailbox_port: int,
+                delegate_port: int,
+                pub_key: str,
+            ):
+                return _make_libp2p_mailbox_connection(
+                    peer_public_key=pub_key, data_dir=".", node_port=mailbox_port
+                )
```

Comment:
I'm confused here. All we do is change the signature and not pass the extra arguments. Why?